Plutarch/DataRepr/Internal/Field.hs

```diff
@@ -100,6 +122,41 @@ instance
   type PFields (PAsData a) = PFields a
   ptoFields = ptoFields . pfromData
 
+
+{- | 
+    always makes sure the result is not wrapped in PAsData, 
+    even if the result it not originally wrapped into it 
+-}
+class PNoAsData (a::PType) (b::PType) | a -> b where 
+  premoveAsData :: Term s a -> Term s b
+
+instance {-# OVERLAPPABLE #-} PNoAsData a a where 
+  premoveAsData = id
+
+instance {-# OVERLAPPABLE #-} a ~ PAsData b => PNoAsData a b  where  
+  premoveAsData = punsafeCoerce
```

Comment:
This isn't legal? It does `premoveAsData :: Term s (PAsData b) -> Term s b`. You want `pfromData`.